File: 卡码/代码随想录/JS/二叉树/01-二叉树遍历方式/LeetCode_0104.js
//104. 二叉树的最大深度
import TreeNode from "../TreeNode.js";

const maxDepth = function(root) {
  q=new Array();
  let depth=0;
  if(root) q.push(root);
  while(q.length>0){
    let size=q.length;
    depth++;
    while(size-->0){
      const node=q.shift();
      if(node.left) q.push(node.left);
      if(node.right) q.push(node.right);
    }
  }
  return depth;
};

File: 卡码/代码随想录/C++/二叉树/01-二叉树遍历方式/LeetCode_0104.cpp
//104. 二叉树的最大深度
#include <iostream>
#include <queue>
#include "../TreeNode.h"
using namespace std;

class Solution {
public:
  int maxDepth(TreeNode* root) {
		int depth=0;
		queue<TreeNode*> q;
		if(root) q.push(root);
		while(!q.empty()){
			int size=q.size();
			depth++;
			while(size-->0){
				TreeNode* node=q.front();
				q.pop();
				if(node->left) q.push(node->left);
				if(node->right) q.push(node->right);
			}
		}
		return depth;
  }
};

int main(){
  system("pause");
  return 0;
}
